cxx/tests/test_misc.cc

```diff
@@ -44,6 +44,18 @@ int main(int argc, char **argv) {
     }
     printf("\n");
 
+    Bigram bg (&prng);
+    bg.incorporate("foo");
+    bg.incorporate("foo");
+    bg.incorporate("_Hello!~");
+    bg.unincorporate("foo");
+    printf("%f\n", exp(bg.logp("bar")));
+    printf("%f\n", exp(bg.logp_score()));
+    for (int i = 0; i < 10; i++) {
+        printf("%s\n", bg.sample().c_str());
```

Comment:
This is fun :) I tried adding
```
    for (int i = 0; i < 300; i++) {
        bg.incorporate("foo");
    }
```
before generation, and it generated
```
foo
fo
fo
fo
foooo
fo
foooo\/Pkv5fo|,C'PbKBn$ELfoo
f$fo
foo
```
so it seems like things are working!

Reply:
Nice!